src/consensus/raft/raft_types.h

```diff
@@ -65,6 +66,13 @@ namespace raft
       if (p)
         p->rollback(v);
     }
+
+    void set_term(Term t)
+    {
+      auto p = x.lock();
```

Comment:
Why does the PBFT equivalent loop until this succeeds, while this version only tries once?

Reply:
I don't know, I've just kept it consistent with the rest of the adaptor, @ashamis?

Reply:
poke @ashamis

Reply:
Sorry just saw this :(

Why does the raft version not spin? x.lock() can fail so the term would then not be set and this happens silently?

Reply:
I'll fix that, even the pbft adapter is doing this inconsistently (see calls up and down).

Reply:
https://github.com/microsoft/CCF/pull/1283